File: src/main/kotlin/me/sd_master92/customvoting/commands/voteparty/VotePartyStartCommand.java
package me.sd_master92.customvoting.commands.voteparty;

import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.constants.Data;
import me.sd_master92.customvoting.subjects.VoteParty;
import me.sd_master92.plugin.command.SimpleSubCommand;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class VotePartyStartCommand extends SimpleSubCommand
{
    private final Main plugin;

    public VotePartyStartCommand(Main plugin)
    {
        super("start");
        this.plugin = plugin;
    }

    @Override
    public void onCommand(CommandSender sender, String[] strings)
    {
        if (!plugin.getData().getLocations(Data.VOTE_PARTY).isEmpty())
        {
            new VoteParty(plugin).start();
        } else
        {
            sender.sendMessage(ChatColor.RED + "There are no registered Vote Party Chests.");
        }
    }

    @Override
    public void onCommand(Player player, String[] strings)
    {

    }
}

File: src/main/kotlin/me/sd_master92/customvoting/commands/voteparty/VotePartyStartCommand.kt
package me.sd_master92.customvoting.commands.voteparty

import me.sd_master92.core.command.SimpleSubCommand
import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.enumerations.PMessage
import me.sd_master92.customvoting.subjects.voteparty.VoteParty
import me.sd_master92.customvoting.subjects.voteparty.VotePartyChest
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class VotePartyStartCommand(private val plugin: CV) : SimpleSubCommand("start")
{
    override fun onCommand(sender: CommandSender, args: Array<String>)
    {
        if (VotePartyChest.getAll(plugin).isNotEmpty())
        {
            if (!VoteParty(plugin).start())
            {
                sender.sendMessage(PMessage.VOTE_PARTY_MESSAGE_QUEUED.toString())
            }
        } else
        {
            sender.sendMessage(PMessage.VOTE_PARTY_ERROR_NO_CHESTS.toString())
        }
    }

    override fun onCommand(player: Player, args: Array<String>)
    {
    }
}